src/components/SignUp/SignUpForm.js

```diff
@@ -0,0 +1,184 @@
+import React, { useState, useCallback, useMemo } from 'react';
+import TextField from '@material-ui/core/TextField';
+import MenuItem from '@material-ui/core/MenuItem';
+import InputLabel from '@material-ui/core/InputLabel';
+import Select from '@material-ui/core/Select';
+import './SignUp.css';
+import Grid from '@material-ui/core/Grid';
+import Button from '@material-ui/core/Button';
+import Airlock from '@calblueprint/airlock';
+
+Airlock.configure({
+  endpointUrl: 'http://localhost:3000',
+  apiKey: 'airlock',
+});
+const BASE_ID = process.env.REACT_APP_AIRTABLE_BASE_ID;
+
+const base = Airlock.base(BASE_ID);
+
+const INITIAL_FORM_STATE = {
+  username: '',
+  fullname: '',
+  password: '',
+  confirmPassword: '',
+};
+
+export default function SignUpForm() {
+  const [formState, setFormState] = useState(INITIAL_FORM_STATE);
+  const [role, setRole] = React.useState('');
+  const [loading, setLoading] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('');
+
+  const roleChange = (event) => {
+    setRole(event.target.value);
+  };
+  const onSubmit = useCallback((event) => {
+    event.preventDefault();
+    setLoading(true);
+    try {
+      /* eslint-disable */
+      // TODO: Figure out if user, token variables need to persist
+      const { user, token } = base.register({
+        username: formState.username,
+        password: formState.password,
+        fields: {
+          user_type: role,
+          display_name: formState.fullname,
+          approval: 'unapproved',
+        },
+      });
```

Comment:
-Make sure to set the state of your inputs back to their defaults and to clear them to be their initial empty state in the client-side part of your app once you submit
-For inspiration on how to do this, check out this feature, specifically the value prop for TextFields:
https://github.com/lablueprint/farm2people-web-app/blob/master/src/components/AddFarm/AddFarm.js